source/config.cpp

```diff
@@ -54,11 +54,14 @@ void Config::read(string const &file_name)
 	string const _namespace_ {"namespace"};
 	string const _function_  {"function"};
 	string const _class_     {"class"};
+	string const _field_     {"field"};
 
 	string const _include_               {"include"};
 	string const _include_for_class_     {"include_for_class"};
 	string const _include_for_namespace_ {"include_for_namespace"};
+	string const _include_before_		 {"include_before"};
```

Comment:
> Could you elaborate why this option is needed? Particularly how is it different from `+include` directive?

All other include directives are written by the code generator *after* <pybind11/pybind11.h>, which includes <Python.h>.

Since I'm creating bindings for a library that makes use of Qt I needed a way to cleanup the preprocessor namespace *before* the inclusion of <pybind11/pybind11.h> in the generated code (Qt defines `slots` for object signal management).

That's the purpose of the `+include_before` directive.

```bash
+include_before <qtcleanup.hpp>
```

```c++
// qtcleanup.hpp
#undef slots
```

Reply:
> All other include directives are written by the code generator after <pybind11/pybind11.h>, which includes <Python.h>.

Hm... this can't be right: all custom includes is always put on top of the file. For example please see test output here: All other include directives are written by the code generator after <pybind11/pybind11.h>, which includes <Python.h>.
https://github.com/RosettaCommons/binder/blob/master/test/T00.basic.ref#L9

This is also explicitly tested here: https://github.com/RosettaCommons/binder/blob/master/test/T31.include_for_class.ref#L2